ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -493,7 +536,11 @@ public object Value
                 }
 
                 var cellValue = HasRichText ? _richText.ToString() : _cellValue;
-                return StringToCellDataType(cellValue);
+                var parsedValue = ParseCellValueFromString(cellValue, _dataType, out String error);
+                if ("" == error)
+                    return parsedValue;
+                else
+                    throw new ArgumentException(error);
```

Comment:
I am not fond of throwing exceptions in getter, especially assuming that the caller do not pass any arguments and can be very surprised with `ArgumentException`. I didn't find the way to get this exception (maybe if a workbook contains errors?) but still this looks dangerous.

Reply:
You're right. I shouldn't be throwing an exception here.